tsl/test/expected/chunk_api.out

```diff
@@ -172,4 +172,141 @@ SELECT * FROM test.show_constraints(format('%I.%I', :'CHUNK_SCHEMA', :'CHUNK_NAM
  constraint_6             | c    | {time}   | -                                         | (("time" >= 'Wed Dec 27 16:00:00 2017 PST'::timestamp with time zone) AND ("time" < 'Wed Jan 03 16:00:00 2018 PST'::timestamp with time zone)) | f          | f        | t
 (4 rows)
 
+TRUNCATE chunkapi;
+-- Create a table with extra columns
+CREATE TABLE extra_col_chunk (time timestamptz NOT NULL, device int, temp float, extra int, CONSTRAINT chunkapi_temp_check CHECK (temp > 0));
+-- Adding a new chunk with extra column should fail
+\set ON_ERROR_STOP 0
+SELECT * FROM _timescaledb_functions.create_chunk('chunkapi', '{"time": [1514419200000000, 1515024000000000]}', NULL, NULL, 'extra_col_chunk');
+ERROR:  table "extra_col_chunk" contains column "extra" not found in parent "chunkapi"
+DETAIL:  The new chunk can contain only the columns present in parent.
+\set ON_ERROR_STOP 1
+-- It should succeed after dropping the extra column
+ALTER TABLE extra_col_chunk DROP COLUMN extra;
+SELECT * FROM _timescaledb_functions.create_chunk('chunkapi', '{"time": [1514419200000000, 1515024000000000]}', NULL, NULL, 'extra_col_chunk');
+ chunk_id | hypertable_id |      schema_name      |    table_name    | relkind |                     slices                     | created 
+----------+---------------+-----------------------+------------------+---------+------------------------------------------------+---------
+        6 |             2 | _timescaledb_internal | _hyper_2_6_chunk | r       | {"time": [1514419200000000, 1515024000000000]} | t
+(1 row)
+
+-- Test creating a chunk with columns in different order
+CREATE TABLE reordered_chunk (device int, temp float, time timestamptz NOT NULL, CONSTRAINT chunkapi_temp_check CHECK (temp > 0));
+SELECT * FROM _timescaledb_functions.create_chunk('chunkapi', '{"time": [1515024000000000, 1515628800000000]}', NULL, NULL, 'reordered_chunk');
+ chunk_id | hypertable_id |      schema_name      |    table_name    | relkind |                     slices                     | created 
+----------+---------------+-----------------------+------------------+---------+------------------------------------------------+---------
+        7 |             2 | _timescaledb_internal | _hyper_2_7_chunk | r       | {"time": [1515024000000000, 1515628800000000]} | t
+(1 row)
```

Comment:
Since this succeeds, do you have a test where you insert into the hypertable parent and verify that the data in the reordered chunk goes into the correct column?

I think it would be sufficient to just insert a single row that will be routed to the chunk and verify that the data is there and is correct.